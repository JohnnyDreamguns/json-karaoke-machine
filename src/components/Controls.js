import React from 'react';

export const Controls = ({
  play, tempo, handleChangeTempo
}) => (
  <React.Fragment>
    <button className="play" data-playing="false" role="switch" aria-checked="false" 
      onClick={play}>Play</button>

    <p>
      <label forhtml="tempo">Tempo</label>
      <input type="range" id="tempo" min="70" max="250" value={tempo} step="1" 
        onChange={handleChangeTempo} />
    </p>
  </React.Fragment>
);

export default Controls;