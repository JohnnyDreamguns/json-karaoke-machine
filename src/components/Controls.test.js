import React from 'react';
import { Controls } from './Controls';
import renderer from 'react-test-renderer';

describe('Controls', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Controls></Controls>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
