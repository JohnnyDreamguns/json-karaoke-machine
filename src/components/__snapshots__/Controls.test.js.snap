// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls renders correctly 1`] = `
Array [
  <button
    aria-checked="false"
    className="play"
    data-playing="false"
    role="switch"
  >
    Play
  </button>,
  <p>
    <label
      forhtml="tempo"
    >
      Tempo
    </label>
    <input
      id="tempo"
      max="250"
      min="70"
      step="1"
      type="range"
    />
  </p>,
]
`;
